boa/network.py

```diff
@@ -89,14 +91,14 @@ class NetworkEnv(Env):
     # always prefetch state in network mode
     _fork_try_prefetch_state = True
 
-    def __init__(self, rpc_url, accounts=None):
+    def __init__(self, rpc: str | RPC, accounts: dict[str, Account] = None):
         super().__init__()
 
-        self._rpc = EthereumRPC(rpc_url)
+        self._rpc = EthereumRPC(rpc) if isinstance(rpc, str) else rpc
```

Comment:
this isn't the end of the world, but it would be best if the constructor didn't do this, especially since i think the `NetworkEnv()` constructor shouldn't really be accessed directly going forward

Reply:
I've added a warning for now